### ISSUE TITLE
GPL-2.0 license is deprecated. 

### ISSUE THREAD

**Issue by sathiyagit** [digitalbazaar/forge#922]
GPL-2.0 license is deprecated.  Can we upgrade latest version GPL-3.0?  I can raise pull request if it is just an update to License file and package.json. 
https://spdx.org/licenses/GPL-2.0.html

**Comment by BenBE** [comment]
IANAL, but AFAIR changing the license requires agreement from all code contributors or removal/replacement of code from contributors that declined the license change. Or you end up with a mess of mixed licensing where parts of the code use one license while the rest uses a completely different one, which would break with the BSD/GPL combination for hopefully obvious reasons.

**Comment by arpitjindal97** [comment]
My understanding is as long as GPL is present none of the MNC will be able to use this library because of the restrictions imposed by it.

**Comment by BenBE** [comment]
> My understanding is as long as GPL is present none of the MNC will be able to use this library because of the restrictions imposed by it.

That's not correct: Anyone could already use it, provided **he complies with the obligations** imposed by the GPL. E.g. if your own project is under GPL (or any other compatible license, like LGPL or AGPL). This BTW doesn't make any difference for GPLv2 vs GPLv3 though and even applies to any license.

**Comment by arpitjindal97** [comment]
Yeah, that's true. but most MNCs have strict guidelines to not use GPL licensed libraries.
Ref: 
 - https://opensource.google/docs/thirdparty/licenses/#banned
 - https://opensource.google/docs/using/agpl-policy/

Recently Grafana also changed their license from Apache to AGPL which caused havoc all around. Companies are looking for an alternative to Grafana.

**Comment by BenBE** [comment]
> Yeah, that's true. but most MNCs have strict guidelines to not use GPL licensed libraries.

Personally, I publish my software under AGPL, so it's hard for me to care less about those MNCs; in particular because I intentionally use AGPL exactly because MNCs hardly share anything back unless you force them. The virality of the GPL is not a flaw, but a feature …

> Recently Grafana also changed their license from Apache to AGPL which caused havoc all around. Companies are looking for an alternative to Grafana.

And that's one of the biggest indicators of the problem: If those companies were contributing back to FLOSS by releasing their derived works for everyone there would be no problem with Grafana being AGPL - and actually no need for it to be. But seeing it to cause havoc means there's just too many freeloaders around that love to spent minimal effort to maximize their effort without giving back to the community.

And in the case of Grafana the issue even was not even actual additional development those companies did, but some minimal tweaking for their paid-for SaaS hosting of Grafana: Making money with the work others did, without paying them. Would have done exactly like the Grafana people did in that situation. ;-)

Well, back on topic: The only reasonable change of license I see for this project is to switch to AGPL, or – in the rare case the developers are fine with commercial use – LGPL. Coming from the GPL side of affairs, BSD makes little to no sense, unless you want a full sell-out of your work.

2ct.

**Comment by davidlehn** [comment]
@sathiyagit Is there a reason you want to update GPL from 2 to 3?

Others:
- I personally think proper etiquette for a license change would be to ask every contributor if it's ok, and we have no resources to do that.
- This project is dual licensed under BSD 3-clause license too. Is that not enough to satisfy most use cases and concerns?
- The licensing was chosen over a decade ago. I'm not sure what the reasoning was at the time. If it were done today we would very likely just use BSD 3-clause license like many of our other projects.
- Licensing is complex and has many tradeoffs and creates strong opinions and views. Please let's not have an open discussion about all that here. :-)  Stay on topic with forge license issues.  Thanks.

**Comment by sathiyagit** [comment]
@davidlehn  - GPL 2 is expired (https://spdx.org/licenses/GPL-2.0.html). Updating to 3 will resolve deprecated licenses issues (Veracode) in our source code.

**Comment by davidlehn** [comment]
@sathiyagit That seems like a strange thing to consider an issue.  There are other projects that use GPL 2.0 only due in part to the changes in 3.0.  Most notably Linux.  Seems like that tool should just accept valid license choices.  And also seems SPDX site should not claim it's "deprecated" but just say there is a newer version, that has important differences, and you might consider using it if it makes sense for you.

To be practical, I doubt we'll be taking the time to make a GPL license version change here anytime soon without a pressing reason since the dual BSD 3 clause license covers most use cases.
